libs/translations/src/translations.service.ts

```diff
@@ -0,0 +1,141 @@
+import { Injectable } from '@nestjs/common';
+import { ConfigService } from '@backend/config';
+import { LoggerService } from '@backend/logger';
+import { LanguageCode, Translation } from './translations-model';
+import { TranslationsRedisService } from './services/translations-redis.service';
+import { crowdinSourceStrings } from './data/source';
+import AdmZip from 'adm-zip';
+import Axios from 'axios';
+import fs from 'fs';
+
+@Injectable()
+export class TranslationsService {
+  constructor(
+    private config: ConfigService,
+    private logger: LoggerService,
+    private translationsRedis: TranslationsRedisService
+  ) {}
+
+  private fileName = 'all.zip';
+
+  public async get(args: {
+    languageCode: string;
+    variables: {
+      name: string;
+    };
+  }) {
+    if (!args.languageCode) {
+      return crowdinSourceStrings;
+    }
+
+    let translations = await this.translationsRedis.getWithExpire();
+
+    if (!translations) {
+      translations = await this.getFromCrowdin();
+      await this.cache(translations);
+    }
+
+    let translation = translations.find(t =>
+      t.languageCodes.includes(args.languageCode)
+    );
+
+    if (!translation) {
+      translation = crowdinSourceStrings;
+    }
+
+    //replace variables with real data (eg. {{name}} => Ali)
+    for (const tKey in translation) {
+      if (translation.hasOwnProperty(tKey) && tKey !== 'languageCodes') {
+        for (const vKey in args.variables) {
+          if (args.variables.hasOwnProperty(vKey)) {
+            translation[tKey] = translation[tKey].replace(
+              `{{${vKey}}}`,
+              `${args.variables[vKey]}`
+            );
+          }
+        }
+      }
+    }
+    return translation;
+  }
+
+  private async getFromCrowdin() {
+    let translations: Translation[];
+    try {
+      await this.buildZips();
+      await this.downloadZip();
+      translations = await this.getNormalizedJson();
+    } catch (err) {
+      translations = await this.translationsRedis.getWithoutExpire();
```

Comment:
1. You should catch only the buildZips part.
2. you should use the exceptions service to report the error.